allspice/utils/bom_generation.py

```diff
@@ -414,3 +452,132 @@ def _group_entries(
+    patch_component_unique_id: dict[str, str] = {}
+    # The keys are the same as above, and the values are a key-value of the
+    # parameter to patch and the value to patch it to.
+    components_to_patch: dict[tuple[str, str], tuple[str, str]] = {}
+
+    for key, value in variant_details.items():
+        # Note that this is in lowercase, as configparser stores all keys in
+        # lowercase.
+        if re.match(r"variation[\d+]", key):
+            variation_details = dict(details.split("=", 1) for details in value.split("|"))
+            try:
+                designator = variation_details["Designator"]
+                # The unique ID field is a "path" separated by backslashes, and
+                # the the unique id we want is the last entry in that path.
+                unique_id = variation_details["UniqueId"].split("\\")[-1]
+                kind = VariationKind(int(variation_details["Kind"]))
+                if kind == VariationKind.NOT_FITTED:
+                    components_to_remove.append((unique_id, designator))
+                else:
+                    patch_component_unique_id[designator] = unique_id
+            except KeyError:
+                logger.warn(
+                    "Designator, UniqueId, or Kind not found in details of variation "
+                    f"{variation_details}"
+                )
+                continue
+            except ValueError:
+                logger.warn(
+                    f"Kind {variation_details['Kind']} of variation {variation_details} must be "
+                    "either 0, 1 or 2."
+                )
+                continue
+        elif re.match(r"paramvariation[\d]+", key):
+            variation_id = key.split("paramvariation")[-1]
+            designator = variant_details[f"ParamDesignator{variation_id}"]
+            variation_details = dict(details.split("=", 1) for details in value.split("|"))
+            try:
+                unique_id = patch_component_unique_id[designator]
+            except KeyError:
+                # This can happen sometimes - Altium allows param variations
+                # even when the component is not fitted, so we just log and
+                # ignore.
+                logger.warn(
+                    f"ParamVariation{variation_id} found for component {designator} either before "
+                    "the corresponding Variation or for a component that is not fitted.\n"
+                    "Ignoring this ParamVariation."
+                )
+                continue
+
+            try:
+                parameter_patch = (
+                    variation_details["ParameterName"],
+                    variation_details["VariantValue"],
+                )
+                if (unique_id, designator) in components_to_patch:
+                    components_to_patch[(unique_id, designator)].append(parameter_patch)
+                else:
+                    components_to_patch[(unique_id, designator)] = [parameter_patch]
```

Comment:
Move this outside of the `try`. You want to guard as small a code block as possible so that it's clear which exception you intend to catch and where it's coming from.

Reply:
Done 👍 Thanks for the callout!